app/validators/csv_keyword_file_validator.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class CsvKeywordFileValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    @file = value
+
+    record.errors.add(attribute, I18n.t('activemodel.csv.errors.invalid_keyword_count')) unless valid_record_count?
+    record.errors.add(attribute, I18n.t('activemodel.csv.errors.invalid_file_type')) unless valid_content_type?
```

Comment:
should we check the type before the `record_count`?

Reply:
oh good one, did not notice that. c20f36211bb1f0c2b72d8a8862b690f328dcea1a